File: src/main/java/chat/teco/tecochat/chat/presentation/chat/socket/ChatSocketConfig.java
package chat.teco.tecochat.chat.presentation.chat.socket;

import chat.teco.tecochat.security.WebSocketHandshakeInterceptor;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
@RequiredArgsConstructor
public class ChatSocketConfig implements WebSocketConfigurer {

    private final ChatSocketHandler chatSocketHandler;
    private final WebSocketHandshakeInterceptor webSocketHandShakeInterceptor;

    @Override
    public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
        registry.addHandler(chatSocketHandler, "/stream/chats/**")
                .addInterceptors(webSocketHandShakeInterceptor)
                .setAllowedOrigins("*");
    }
}

File: src/main/kotlin/chat/teco/tecochat/config/ChatSocketConfig.kt
package chat.teco.tecochat.config

import chat.teco.tecochat.security.WebSocketHandshakeInterceptor
import chat.teco.tecochat.ui.ChatSocketHandler
import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry


@EnableWebSocket
@Configuration
class ChatSocketConfig(
    private val chatSocketHandler: ChatSocketHandler,
    private val webSocketHandShakeInterceptor: WebSocketHandshakeInterceptor,
) : WebSocketConfigurer {

    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry.addHandler(chatSocketHandler, "/stream/chats/**")
            .addInterceptors(webSocketHandShakeInterceptor)
            .setAllowedOrigins("*")
    }
}

